rasterio/_base.pyx

```diff
@@ -449,6 +450,48 @@ cdef class DatasetBase(object):
                     GDALGetRasterUnitType(self.band(j)) for j in self.indexes)
             return self._units
 
+    def block(self, bidx, i, j):
```

Comment:
One minor comment: `Dataset.block()` sounds like a verb to me, and in isolation a user might expect it do something rather different. Perhaps `block_info()` would be less ambiguous.